### PR TITLE
Fix "Avoid mutable default arguments" issue

### PR DESCRIPTION
**Pull request by renemilk**
This pull request automatically fixes all occurrences of the following issue:

Issue type: [Avoid mutable default arguments](https://www.quantifiedcode.com/app/issue_class/3P0qV6OB)
Issue details: [https://www.quantifiedcode.com/app/project/gh:pymor:pymor?groups=code_patterns/%3A3P0qV6OB](https://www.quantifiedcode.com/app/project/gh:pymor:pymor?groups=code_patterns/%3A3P0qV6OB)

To adjust the commit message or the actual code changes, just [rebase](https://git-scm.com/docs/git-rebase) or [cherry-pick](https://git-scm.com/docs/git-cherry-pick) the commits.
For questions or feedback reach out to cody@quantifiedcode.com.

Legal note: We won't claim any copyrights on the code changes.

Cheers,
[Cody - Your code quality bot](https://www.quantifiedcode.com/cody)

### CODE REVIEW

File: src/pymor/core/logger.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def getChild(self):
 
 
 @defaults('levels', sid_ignore=('levels',))
-def set_log_levels(levels={'pymor': 'INFO'}):
+def set_log_levels(levels=None):
```
Comment (sdrave) [inline src/pymor/core/logger.py]: This change would break pyMOR's defaults handling.

File: src/pymor/operators/numpy.py [hunk 2]
```diff
@@ -469,7 +469,7 @@ def sparse_options(default_solver='spsolve',
                    pyamg_sa_smooth=('jacobi', {'omega': 4.0/3.0}),
                    pyamg_sa_presmoother=('block_gauss_seidel', {'sweep': 'symmetric'}),
                    pyamg_sa_postsmoother=('block_gauss_seidel', {'sweep': 'symmetric'}),
-                   pyamg_sa_improve_candidates=[('block_gauss_seidel', {'sweep': 'symmetric', 'iterations': 4}), None],
+                   pyamg_sa_improve_candidates=None,
```
Comment (sdrave) [inline src/pymor/operators/numpy.py]: See above.

**Comment by sdrave** [comment]
Two of the changes introduce bugs in pyMOR. Regarding the other two: I'm not sure if they really increase the quality of the code. Other opinions?

**Comment by renemilk** [comment]
Do not like either one. Bounding box could be tuples, probably. The Auto-Fix should better use one line conditional assignment rather than if blocks, for more concise code.

**Pull request closed without merging**